### PR TITLE
Annotations

### PR DESCRIPTION
**Pull request by chrisspre**

### CODE REVIEW

File: tools/rsdl/rapid-cli/rapid/samples/demo/sample.rsdl [hunk 1]
```diff
@@ -19,26 +19,26 @@ type Company
...
 {
-    @key id: Integer
-    key: Boolean
+    key id: Integer
+
     name : Name
     title: String
     employmentType: EmploymentType
 
     // a RSDL function that will be translate to a CSDL function
```
Comment (ralfhandl) [inline tools/rsdl/rapid-cli/rapid/samples/demo/sample.rsdl:28]:
```suggestion
    // a RSDL function that will be translate to a CSDL action
```

File: tools/rsdl/syntax-highlighting/syntaxes/rsdl.tmLanguage.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
       "patterns": [
         {
           "name": "storage.type",
-          "match": "\\b(type|enum|service)\\b"
+          "match": "\\b(type|enum|service|namespace|include|as)\\b"
```
Comment (ralfhandl) [inline tools/rsdl/syntax-highlighting/syntaxes/rsdl.tmLanguage.json:22]:
```suggestion
          "match": "\\b(type|enum|flags|service|namespace|include|as)\\b"
```

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 3]
```diff
@@ -37,51 +37,71 @@ modelElement = structuredType / enumType / service
...
 ```
 
 ### Enumeration Types
 
 ```ABNF
-enumType   = ( %s"enum" / $s"flags" ) identifier "{" 1*enumMember "}"
+enumType             = annotations ( %s"enum" / $s"flags" ) identifier "{" 1*enumMember "}"
 
-enumMember = identifier
+enumMember           = identifier
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:68]:
```suggestion
enumMember           = annotations identifier
```

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 4]
```diff
@@ -37,51 +37,71 @@ modelElement = structuredType / enumType / service
...
 
-operation            = [ actionAnnotation ] identifier
+operation            = annotations [operationModifier] identifier
                        "(" [ parameter *("," parameter) ] ")"
                        [ ":" typeReference ]
 
-actionAnnotation     = %s"@action"
+operationModifier    = %s"action" / %s"function"
 
 parameter            = identifier ":" typeReference
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:60]:
```suggestion
parameter            = annotations identifier ":" typeReference
```

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 5]
```diff
@@ -37,51 +37,71 @@ modelElement = structuredType / enumType / service
...
 
-enumMember = identifier
+enumMember           = identifier
 ```
 
 ### Service
 
 ```ABNF
-service          = %s"service" "{" 1*serviceMember "}"
+service              = %s"service" "{" 1*serviceMember "}"
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:74]:
```suggestion
service              = annotations %s"service" "{" 1*serviceMember "}"
```

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 6]
```diff
@@ -37,51 +37,71 @@ modelElement = structuredType / enumType / service
...
 
-serviceMember    = entitySet / singleton / serviceOperation
+serviceMember        = entitySet / singleton / serviceOperation
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:76]:
```suggestion
serviceMember        = annotations ( entitySet / singleton / serviceOperation )
```

**Review by ralfhandl — COMMENTED** [review]
See suggestions

File: docs/rsdl/rapid-pro-rsdl-abnf.md (hunk 3 above)
Comment (chrisspre) [reply docs/rsdl/rapid-pro-rsdl-abnf.md:68]: would like to address this separately in issue #176 . Please resolve if you agree.

File: docs/rsdl/rapid-pro-rsdl-abnf.md (hunk 4 above)
Comment (chrisspre) [reply docs/rsdl/rapid-pro-rsdl-abnf.md:60]: would like to address this separately in issue #176 . Please resolve if you agree.

File: docs/rsdl/rapid-pro-rsdl-abnf.md (hunk 5 above)
Comment (chrisspre) [reply docs/rsdl/rapid-pro-rsdl-abnf.md:74]: would like to address this separately in issue #176 . Please resolve if you agree.

File: docs/rsdl/rapid-pro-rsdl-abnf.md (hunk 6 above)
Comment (chrisspre) [reply docs/rsdl/rapid-pro-rsdl-abnf.md:76]: would like to address this separately in issue #176 . Please resolve if you agree.

**Pull request merged**